src/logic/map_objects/immovable.h

```diff
@@ -170,6 +170,8 @@ class ImmovableDescr : public MapObjectDescr {
 		became_from_.insert(s);
 	}
 
+	Descriptions& descriptions_;
```

Comment:
Don't expose the attribute to the public, only the required functions.
It looks like you only need `add_immovable_relation`